pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1175,21 +1175,18 @@ func (m *MacvtapPodInterface) discoverPodNetworkInterface() error {
 	}
 	m.podNicLink = link
 
-	if len(m.vif.MAC) == 0 {
```

Comment:
AFAIU, you cannot remove this, since I don't think your claim "There is no scenario we get to "discoverPodNetworkInterface" and the cache already contains the mac." is right; when the user specifies the MAC address of the interface, that gets written to the `vif.MAC` attribute, and is done on phase 1 (executed on virt-handler).

We do that check in order to assure the user specified MAC is kept.

https://github.com/kubevirt/kubevirt/blob/0454b350570531c9562fc18d4c576ae1910e6cde/pkg/virt-launcher/virtwrap/network/podinterface.go#L184

https://github.com/kubevirt/kubevirt/blob/0454b350570531c9562fc18d4c576ae1910e6cde/pkg/virt-launcher/virtwrap/network/podinterface.go#L315

Reply:
You're right! I missed the fact vif.mac is (ab)used to keep the macadress from the vm api.
Changed the code so that for macvtap `virtinterface` will keep this address.

Reply:
Ok, good for me.